CMSgov/mint-app
# [EASI-3067] - PDF/CSV readonly export

# EASI-3067

## Changes and Description

[Figma](https://www.figma.com/file/6tkIhfNfcsKEAkqXo1qvXH/MINT---Read-Only-View?type=design&mode=design&t=vz4EqajLNTcwzeeg-0)

Implemented [react-to-print](https://www.npmjs.com/package/react-to-print) package for using the browser's native print api

There is something we may want to discuss as a group further.  In order to appease conflicted font-family settings, we have a `@print` override that get set when print window opens.  This inadvertently affects the background text elements.  We could*** have this font overwride on the entire app, but we arent quite sure of the global consequences.  This would fix the app looking wonky when the print window is open.

Share/Export nav is functional, but no Share yet, both will render same component

- Modal to initiate a download of a model plan for pdf/csv export
- Link on task list side nav for export
- Dropdown selection for a filtered view export
  - Default to filter view if currently viewing a group
  - CSV will be disabled for all filter views other than the componsite readonly
  - Default to all model plans if task list
- Checkbox selection of download file type
- Button on readonly banner for export
- Extracted summarybox component to insert on all pdfs
- All links some header info should be stripped out of pdf
- Extracted csv fetch and download functionality into own hook to be used in varying places

Additional Changes: 

- For testing purposes, created a central mock file to be used for all of readonly/export testing
- Had to add an event listener on the app `index.tsx` to wait for dom to be loaded for app to run
  - This was an unknown cause, but the solution fixed it and doesn't seem to have any real app effect other than to appease the testing suite




![Screenshot 2023-07-18 at 2 04 56 PM](https://github.com/CMSgov/mint-app/assets/95709965/f7ca3acd-593e-435a-93f5-1008530a1bb5)

## How to test this change

- Navigate to task list or readonly
- Click the button/link to export/share model plan
- Verify that if viewing a filtered view, that option autopopulates
- CSV should only be selectable if all model plan information is selected
- Button to export should be disabled if relevant conditions arent met
- Tests should run successfully


## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
